crates/net/downloaders/src/headers/reverse_headers.rs

```diff
@@ -591,15 +652,25 @@ where
 
                     // Update the sync target hash
                     self.sync_target = match self.sync_target.take() {
-                        Some(mut sync_target) => {
-                            sync_target.hash = target;
-                            Some(sync_target)
-                        }
+                        Some(sync_target) => Some(sync_target.with_hash(target)),
                         None => Some(SyncTargetBlock::from_hash(target)),
                     };
                     self.on_block_number_update(parent_block_number, parent_block_number);
                 }
             }
+            SyncTarget::TipNum(num) => {
+                let current_tip_num = self.sync_target.as_ref().and_then(|t| t.number());
+                if Some(num) != current_tip_num {
+                    trace!(target: "downloaders::headers", %num, "Updating sync target based on num");
+                    self.metrics.out_of_order_requests.increment(1);
```

Comment:
is this the right metric?

Reply:
I guess not - it doesn't really represent the situation that the metric is trying to measure, would not be a useful metric because the continuous downloader is 1-by-1, and this match arm is not used anywhere else. It would just report `n = number of headers` out of order requests. removed it